buildSrc/src/main/java/Gecko.kt

```diff
@@ -0,0 +1,11 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+private object GeckoVersions {
```

Comment:
We may want to consider renaming this to `EngineVersions` and keeping this version of the file in the Gecko source set.

Reply:
@Baron-Severin Hmm, how come? We'd only maintain gecko since we have no control over system webview versioning (I can see us adding different CPU and channel variants in the future though)

Reply:
We made pretty heavy use of source sets at Macy's (the Bloomingdale's app was just a different build flavor), and we found that the complexity involved tended to compound really quickly.

Gecko is a source set, and to me that means that Gecko related code should never be accessible to System code.  If we wind up with a lot of conditional checks to execute System vs Gecko code, things get messy and complicated very fast.  I strongly feel that that kind of problem should be handled using source sets as often as possible.

In this case, the System version of the class might just query a `WebView` and gets its version, or something similar.  But I think we should strictly enforce a policy where Main code knows about Main concerns and nothing else.

Reply:
@Baron-Severin Just to clarify, this is a `/buildSrc/` that works specifically with declarative portion of `build.gradle`, in other words it's not an actual source set? Main or build variant source codes has no direct exposure to this

Reply:
I totally missed that, sorry.  Ignore me!